File: android/app/src/main/java/com/qiwu/test/widget/emoji/EmojiReactTextView.java
package com.qiwu.test.widget.emoji;

import android.annotation.SuppressLint;
import android.content.Context;
import android.text.InputFilter;
import android.view.ActionMode;
import android.widget.TextView;

import androidx.core.widget.TextViewCompat;
import androidx.emoji.widget.EmojiTextViewHelper;

import com.facebook.react.views.text.ReactTextView;

public class EmojiReactTextView extends ReactTextView {

    private EmojiTextViewHelper mEmojiTextViewHelper;

    /**
     * Prevent calling {@link #init()} multiple times in case super() constructors
     * call other constructors.
     */
    private boolean mInitialized;

    public EmojiReactTextView(Context context) {
        super(context);
        init();
    }

    private void init() {
        if (!mInitialized) {
            mInitialized = true;
            getEmojiTextViewHelper().updateTransformationMethod();
        }
    }

    @Override
    public void setFilters(InputFilter[] filters) {
        super.setFilters(getEmojiTextViewHelper().getFilters(filters));
    }

    @Override
    public void setAllCaps(boolean allCaps) {
        super.setAllCaps(allCaps);
        getEmojiTextViewHelper().setAllCaps(allCaps);
    }

    private EmojiTextViewHelper getEmojiTextViewHelper() {
        if (mEmojiTextViewHelper == null) {
            mEmojiTextViewHelper = new EmojiTextViewHelper(this);
        }
        return mEmojiTextViewHelper;
    }

    /**
     * See
     * {@link TextViewCompat#setCustomSelectionActionModeCallback(TextView, ActionMode.Callback)}
     */
    @SuppressLint("RestrictedApi")
    @Override
    public void setCustomSelectionActionModeCallback(ActionMode.Callback actionModeCallback) {
        super.setCustomSelectionActionModeCallback(TextViewCompat
                .wrapCustomSelectionActionModeCallback(this, actionModeCallback));
    }
}

File: android/lib/src/main/java/com/qiwu/lib/widget/emoji/EmojiReactTextView.kt
package com.qiwu.lib.widget.emoji

import android.annotation.SuppressLint
import android.content.Context
import android.text.InputFilter
import android.view.ActionMode
import androidx.core.widget.TextViewCompat
import androidx.emoji.widget.EmojiTextViewHelper
import com.facebook.react.views.text.ReactTextView

class EmojiReactTextView(context: Context) : ReactTextView(context) {

    private var mEmojiTextViewHelper: EmojiTextViewHelper? = null

    /**
     * Prevent calling [.init] multiple times in case super() constructors
     * call other constructors.
     */
    private var mInitialized = false

    init {
        if (!mInitialized) {
            mInitialized = true
            getEmojiTextViewHelper().updateTransformationMethod()
        }
    }

    override fun setFilters(filters: Array<InputFilter?>?) {
        super.setFilters(getEmojiTextViewHelper().getFilters(filters!!))
    }

    override fun setAllCaps(allCaps: Boolean) {
        super.setAllCaps(allCaps)
        getEmojiTextViewHelper().setAllCaps(allCaps)
    }

    private fun getEmojiTextViewHelper(): EmojiTextViewHelper {
        if (mEmojiTextViewHelper == null) {
            mEmojiTextViewHelper = EmojiTextViewHelper(this)
        }
        return mEmojiTextViewHelper!!
    }

    /**
     * See
     * [TextViewCompat.setCustomSelectionActionModeCallback]
     */
    @SuppressLint("RestrictedApi")
    override fun setCustomSelectionActionModeCallback(actionModeCallback: ActionMode.Callback?) {
        super.setCustomSelectionActionModeCallback(
            TextViewCompat
                .wrapCustomSelectionActionModeCallback(this, actionModeCallback!!)
        )
    }
}